### DISCUSSION TITLE
create-your-own-website-part2/

### DISCUSSION THREAD

**Discussion by giscus** [amir-pourmand/hugo-papermod-farsi-template#30] [Announcements]
# create-your-own-website-part2/

قبلاً راجع به ساخت وب‌سایت استاتیک مطلب کوتاهی نوشته بودم و گفته بودم که به زودی تکمیل‌اش می‌کنم. برای این که روضه‌هایی که خوندم رو مجدداً تکرار نکنم ابتدا این پست رو بخونید.
آموزش ساخت وب‌سایت شخصی استاتیک - قسمت یک | گاه‌نوشته‌های امیرپورمند
بیشتر که فکر کردم دیدم که بهتره این دست مطالب رو با ویدئو توضیح بدم و یک ویدئو ضبط کردم. متأسفانه ویدئو نسبتاً طولانی شد (۱ ساعت) و به نظرم میشد ویدئو رو در نیم ساعت هم ضبط کردم ولی گفتم اگر کمال‌گرایی از خودم دربیارم، احتمالاً این مطلب هم مدت‌ها عقب بیفته.

https://aprd.ir/create-your-own-website-part2/

**Comment by Davaji** [comment]
سلام. اگه بخواهیم این وبلاگ رو روی شرکت‌های هاستینگ داخلی هم بالا بیاریم امکان پذیره؟ به دلیل همون قضیه‌ی فیلترینگ که اشاره کردید می‌پرسم.

**Reply by pourmand1376** [reply]
سلام. بله که میشه. منتهی یک تفاوتی که هست اینه که معمولاً شرکت‌های هاستینگ ایرانی دقیقا Workflowای که گفتم رو پشتیبانی نمی‌کنند؛ یعنی قرار نیست به گیت وصل بشن و خودشون برن سایت Hugo رو براتون بسازند. 

به احتمال بسیار زیاد این تیکه‌اش رو باید خودتون انجام بدید. یعنی بعد از این که تغییری در سایت دادید، خودتون بصورت لوکال سایت رو Build کنید و بفرستید رو سرور اونها. چون خروجی Hugo نهایتاً یه سری فایل Html css هست به راحتی می‌تونند Serveاش کنند و مشکلی از این جهت نیست.

**Comment by brezaie** [comment]
سلام امیر جان؛
روزت بخیر

با توجه به آپدیت Hugo، برخی از meta dataها، مثل تاریخ به درستی نمایش داده نمی‌شوند. برای همین موضوع، نیاز هست تغییری در فایل post_meta.html ایجاد بشه بصورت زیر:

https://github.com/adityatelange/hugo-PaperMod/commit/7795c90f6fa106733267481d9ed0518b00f2c62c

ممنونم از راهنمایی خیلی خوبی که برای راه‌اندازی یک وبلاگ ساده و جذاب داشتی.

**Comment by Abmokhtari** [comment]
سلام امیر عزیز،  
متشکرم از زحمتی که برای فیلم کشیدی راستش خیلی من دوست دارم سایت استاتیک بیارم بالا اما چیزهایی که در بالا توضیح دادید انگار دانش بشتری میخواد حداقل برای من و ترتیب کار رو برای منی که زیاد با گیت و دستور فرمان پاورشل رو بلد نیستم سخت شد. خواستم ببینم لینک دیگری نداری مثلا در یوتیوب یا هر آموزش دیگری که خودت تشخیص بدی خوبه که برای من و امثال من که مبتدی هستند در بحث گیت، یه آموزش مرحله به مرحله باشه که بتونیم به ترتیب انجام بدیم و سایتمون رو بالا بیاریم؟ 
بسیار از شما متشکرم

**Reply by pourmand1376** [reply]
سلام. وقتت بخیر. 

راستش چند وقتی صبر کردم که خودم یه آموزش ساده‌تر ضبط کنم اما فرصت نشد. 

به نظرم این کار رو یک جا انجام نده. این که وب‌سایتی مثل من بالا بیاری و دامنه رو بهش وصل کنی و کلی فیچر دیگه رو همون اول داشته باشی، خیلی کار می‌بره. 

مثلاً اول Github Pages رو یاد بگیر. 

- https://www.youtube.com/watch?v=5XhxR9Vs6zc
- https://www.youtube.com/watch?v=srQjolk7rj8

بعد می‌تونی یه سری تمپلت پیدا کنی و رزومه‌ات رو با تغییرات مینیمال روی گیت هاب بالا بیاری. چیزی مثل این:
- https://github.com/sharu725/online-cv

بعد به مرور می‌تونی یاد بگیری Hugo چیه و چه روش‌هایی میشه سایت استاتیک بالا آورد. کم‌کم از Github Pages هم می‌تونی فاصله بگیری و سراغ Cloudflare و روش‌های پیچیده‌تر بری. شاید بعدش هم دلت خواست دامنه‌ات رو بهش وصل کنی و سیستم کامنت گذاری هم در آخرین مرحله باید اضافه بشه. 

اذیت میشی اگر بخوای یهو بپری مرحله آخر.
